src/CoreEnforcer.php

```diff
@@ -354,19 +354,32 @@ public function clearPolicy(): void
      */
     public function loadPolicy(): void
     {
-        $this->model->clearPolicy();
+        $flag = false;
+        $needToRebuild = false;
+        $oldModel = $this->model;
+        $this->model = clone $oldModel;
+        $this->clearPolicy();
 
         try {
             $this->adapter->loadPolicy($this->model);
         } catch (InvalidFilePathException $e) {
-            //throw $e;
+            $flag = true;
         }
 
         $this->model->printPolicy();
         $this->model->sortPoliciesByPriority();
+
         if ($this->autoBuildRoleLinks) {
+            $needToRebuild = true;
             $this->buildRoleLinks();
         }
+
+        if ($flag) {
+            $this->model = $oldModel;
+            if ($this->autoBuildRoleLinks && $needToRebuild) {
+                $this->buildRoleLinks();
+            }
+        }
```

Comment:
@basakest First, when an exception is caught, an exception should be thrown. Before throwing an exception, do some rollback operations. The general process is as follows :
```php
        // Todo sth.
        $newModel = clone $this->model;
        try {
            $this->adapter->loadPolicy($this->model);

            // Todo some things after load success...

        } catch (InvalidFilePathException $e) {
            // Ignore throw $e;
        } catch (\Throwable $e){
            throw $e;
        }finally {
            // Todo some things after load failed...
        }
        // ...
```